Support `input[type="range"]` in sideways-lr writing mode

nt1m opened the pull request:
#### b1c49c2d197b012cf591a97289eec7bae8804a87
<pre>
Support `input[type=&quot;range&quot;]` in sideways-lr writing mode
<a href="https://bugs.webkit.org/show_bug.cgi?id=285571">https://bugs.webkit.org/show_bug.cgi?id=285571</a>
<a href="https://rdar.apple.com/142518818">rdar://142518818</a>

Reviewed by Aditya Keerthi.

Painting already works properly.

To fix dragging the slider, use `WritingMode::isLogicalLeftInlineStart()` instead of checking `isLeftToRightDirection()` / `isBidiLTR()`.

* LayoutTests/fast/forms/range/sideways-lr-dragging-expected.txt: Added.
* LayoutTests/fast/forms/range/sideways-lr-dragging.html: Added.
* LayoutTests/platform/ios/TestExpectations:
* Source/WebCore/html/shadow/SliderThumbElement.cpp:
(WebCore::SliderThumbElement::setPositionFromPoint):
* Source/WebCore/platform/graphics/controls/SliderTrackPart.cpp:
(WebCore::SliderTrackPart::drawTicks const):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::paintSliderTicks):
* Source/WebCore/rendering/ios/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::paintSliderTicks):

Canonical link: <a href="https://commits.webkit.org/288752@main">https://commits.webkit.org/288752@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/78c35495fa0b8c6df073f28ac9eb19efb6c0f9b1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/84317 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/3941 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/38622 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/89395 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/35326 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/86402 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/4027 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/11916 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/65575 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/23409 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/87363 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/3020 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/76609 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/45868 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/2971 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/30839 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/34375 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/73866 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/31607 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/90776 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/11584 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/8413 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/74027 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/11811 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/72434 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/73225 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18119 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/17552 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/15996 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/2949 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/11536 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/17012 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/11385 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/14861 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/13158 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on LayoutTests/fast/forms/range/sideways-lr-dragging.html:
```diff
@@ -0,0 +1,40 @@
+                eventSender.mouseMoveTo(slider.offsetLeft + 7, slider.offsetTop + slider.offsetHeight - 7);
+                eventSender.mouseDown();
+                eventSender.mouseMoveTo(slider.offsetLeft + 7, slider.offsetTop + 7);
+                eventSender.mouseUp();
```
Should be able to get this test working on iOS as well if you use `UIHelper.dragFromPointToPoint`.

nt1m replied:
Done

pxlcoder reviewed (approved):
r=mews

nt1m committed:
Support `input[type="range"]` in sideways-lr writing mode
https://bugs.webkit.org/show_bug.cgi?id=285571
rdar://142518818

Reviewed by Aditya Keerthi.

Painting already works properly.

To fix dragging the slider, use `WritingMode::isLogicalLeftInlineStart()` instead of checking `isLeftToRightDirection()` / `isBidiLTR()`.

* LayoutTests/fast/forms/range/sideways-lr-dragging-expected.txt: Added.
* LayoutTests/fast/forms/range/sideways-lr-dragging.html: Added.
* LayoutTests/platform/ios/TestExpectations:
* Source/WebCore/html/shadow/SliderThumbElement.cpp:
(WebCore::SliderThumbElement::setPositionFromPoint):
* Source/WebCore/platform/graphics/controls/SliderTrackPart.cpp:
(WebCore::SliderTrackPart::drawTicks const):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::paintSliderTicks):
* Source/WebCore/rendering/ios/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::paintSliderTicks):

Canonical link: https://commits.webkit.org/288752@main
